docs/src/model_simulation/examples/activation_time_distribution_measurement.md

```diff
@@ -0,0 +1,58 @@
+
+Next, we wish to measure the distribution of these activation times. First we will create a [callback](https://docs.sciml.ai/DiffEqDocs/stable/features/callback_functions/) which terminates the simulation once it has reached a threshold. This both ensures that we do not have to expend unnecessary computer time on the simulation after its activation, and also enables us to measure the activation time as the final time point of the simulation. Here we will use a [discrete callback](https://docs.sciml.ai/DiffEqDocs/stable/features/callback_functions/#SciMLBase.DiscreteCallback). By looking at the previous simulations, we determine $X = 100$ as a suitable activation threshold. We use the `terminate!` function to terminate the simulation once this value has been reached.
+```@example activation_time_distribution_measurement
+condition(u, t, integrator) = integrator[:X] > 100.0
+affect!(integrator) = terminate!(integrator)
+callback = DiscreteCallback(condition, affect!)
+nothing # hide
+```
+Next, we will perform our ensemble simulation. By default, for each simulation, these save the full trajectory. Here, however, we are only interested in the activation time. This enables us to utilise an [*output function*](https://docs.sciml.ai/DiffEqDocs/dev/features/ensemble/#Building-a-Problem). This will be called at the end of every single simulation and determine what to save (it can also be used to potentially rerun individual simulations, however, we will not use this feature here). Here we create an output function which saves only the simulation's final time point. We also make it throw a warning if the simulation reaches the end of the simulation time frame ($t = 2000$) (this indicates a simulation that never activated, the occurrences of such simulation would cause us to underestimate the activation times). Finally, just like previously, we must set `safetycopy = true`.
+```@example activation_time_distribution_measurement
+function output_func(sol, i)
+    (sol.t[end] == tspan[2]) && @warn "A simulation did not activate during the given time span."
+    return (sol.t[end], false)
+end
+eprob = EnsembleProblem(sprob; output_func, safetycopy = true)
+esol = solve(eprob, ImplicitEM(); trajectories = 250, callback)
+nothing # hide
+```
+Finally, we can plot the distribution of activation times. For this, we will use the [StatsPlots.jl](https://docs.juliaplots.org/latest/generated/statsplots/) package's `density` function (essentially a smoothed histogram). The input to `density` is the activation times (which our output function has saved to `esol.u`).
+```@example activation_time_distribution_measurement
+using StatsPlots
+density(esol.u; label = "Activation time distribution", xlabel = "Activation time")
+```
```

Comment:
Why not just use a normal histogram with the option to have it normalize to a pdf? That shows the actual sample noise in the bins instead of smoothing everything out, and avoids the StatsPlots dependency.

Reply:
I generally have preferred to use `density` as I feel those plots are better and easier to work with. But if avoiding `StatsPlots` is priority I can plot a histogram and then link to the `desnity` function suggesting that readers should try it out as well?